1.8/tutorial/app2.md

```diff
@@ -0,0 +1,59 @@
+---
+post_title: DC/OS 101 - User Facing Apps
+menu_order: 3
+---
+
+# Prerequisites
+We expect you to have access to a running DC/OS OSS cluster and the DC/OS CLI installed and configured.
+Furthermore, we expect you have just deployed app1.
+
+# Objective
+We already deployed an app which is running internally in our cluster (i.e., it is not targeted to users directly). Next, will deploy an app which provides a web UI to users.
+Secondly, we also want to deploy an app natively, i.e., not relying on Docker.
+
+
+# Steps
+  * Check out app1
+    * Take a short look at out [app2](https://github.com/joerg84/dcos-101/blob/master/app2/app2.go). App2 is a go-lang based HTTP server exposing a very simple interface to redis.
+  * Deploy app
+    * Take a short look at the [app definition](https://raw.githubusercontent.com/joerg84/dcos-101/master/app2/app2.json). In this case, the app is a binary without external dependencies.
+    Hence there is no need to deploy it in a docker container.
+    * Add app `dcos marathon app add https://raw.githubusercontent.com/joerg84/dcos-101/master/app2/app2.json`.
+  * Let us check that app2 is running
+    * By looking at all DC/OS tasks: `dcos task`
+    * By looking at all marathon apps: `dcos marathon app list`
+    * Curl the http server from within cluster (in this case from the leading master):
+       * `dcos node ssh --master-proxy --leader`
+       * `curl dcos-101app2.marathon.l4lb.thisdcos.directory:10000`
+       This should return you dome raw html code from app2's webserver.
+    * Make it available to the public
+      Curling the app from within the cluster and viewing the raw html is a nice thing (more or less), but in reality, we want to expose the app to the public. Therefore we have understood that DC/OS has two different [node types](https://docs.mesosphere.com/1.8/overview/concepts/#dcos-agent-node): private and public agent nodes. Private agent nodes (usually) do not have ingress access from outside of the cluster, while public agent nodes allow for ingress access from outside the cluster.
+      Marathon will start application and services by default on private agent nodes and hence they cannout be accessed from the outside. The default pattern to expose an app to the outside is to use a load-balancer running on one of the public nodes. We will revisit the topic of load-balancing later in this tutorial, but for now, we choose marathon-lb as one potential load-balancer.
```

Comment:
By default, Marathon will start applications and services on private agent nodes, which prevents them from being accessed from outside the cluster. To expose an app to the outside we usually use a load-balancer running on one of the public nodes. We will revisit the topic of load-balancing (and load-balancer options) later in this tutorial, but for now, we choose marathon-lb as our load-balancer.